src/math_parser_diag.cpp

```diff
@@ -1321,6 +1321,55 @@ std::function<double( dialogue & )> vision_range_eval( char scope,
     };
 }
 
+std::function<double( dialogue & )> calories_eval( char scope,
+        std::vector<diag_value> const &/* params */, diag_kwargs const &kwargs )
+{
+    diag_value format_value( std::string( "raw" ) );
+    if( kwargs.count( "format" ) != 0 ) {
+        format_value = *kwargs.at( "format" );
+    }
+
+    return[format_value, beta = is_beta( scope )]( dialogue const & d ) {
+        std::string format = format_value.str( d );
+        if( format != "raw" && format != "percent" ) {
+            debugmsg( R"(Unknown format type "%s" for calories, assumning "raw")", format );
+            format = "raw";
+        }
+
+        if( format == "percent" ) {
+            if( d.actor( beta )->get_character() ) {
+                int divisor = d.actor( beta )->get_healthy_kcal() / 100;
+                //if no data, default to default height of 175cm
+                if( divisor == 0 ) {
+                    divisor = 118169 / 100;
```

Comment:
Where does the 118169 magic number come from? Either use the calculation that generates it or just have a debugmsg here and give some safe value, not a (liable to change) calculated value.

Reply:
right from the original `stored_kcal_percentage` implementation
https://github.com/CleverRaven/Cataclysm-DDA/pull/73136/files#diff-534f433e7a0b5ec5c233390037e131f3717e8aa5f347c47e650a65a1ddc95279L2133

Reply:
Sorry, that was a rhetorical question. There shouldn't be a magic number like this.